torch/csrc/lazy/ts_backend/ts_node.cpp

```diff
@@ -27,11 +27,7 @@ const Shape& GetShapeFromTsValue(const Value& value) {
 
 void TsNodeSetShapeDeferred(
     NodePtr node, const std::function<Shape()>& shape_fn) {
-  if (auto tsnode = std::dynamic_pointer_cast<TsNode>(node)) {
-    tsnode->SetShapeDeferred(shape_fn);
-    return;
-  }
-  throw std::runtime_error("Expected TsNode but could not dynamic cast");
+  throw std::runtime_error("SetShapeDeferred is not supported. IRs calling this aren't used anyway");
```

Comment:
why can't we remove it then?

Reply:
yea, it should be removed. I had a final commit that removed it and also broke something else, so i just dropped that one and moved to land the working part first